File: jacobi.R
jacobi <- function(A, b, x0 = NULL, tol = 1e-6, max_iter = 100) {
  n <- nrow(A)
  if (is.null(x0)) x0 <- rep(0, n)
  x_new <- x0
  cat("Iteración 0:", x_new, "\n")
  for (k in 1:max_iter) {
    x_old <- x_new
    for (i in 1:n) {
      s <- sum(A[i, -i] * x_old[-i])
      x_new[i] <- (b[i] - s) / A[i, i]
    }
    cat("Iteración", k, ":", round(x_new, 6), "\n")
    if (sqrt(sum((x_new - x_old)^2)) < tol) break
  }
  return(x_new)
}

File: amapp/metodos/jacobi.py
def jacobi_con_pasos(A, b, tol=1e-10, max_iter=100):
    n = len(A)
    x = [0]*n
    pasos = []
    for it in range(max_iter):
        x_new = [0]*n
        for i in range(n):
            s = sum(A[i][j]*x[j] for j in range(n) if j != i)
            x_new[i] = (b[i] - s) / A[i][i]
        pasos.append(f"Iteración {it+1}: {x_new}")
        if max(abs(x_new[i]-x[i]) for i in range(n)) < tol:
            break
        x = x_new
    pasos.append(f"Solución final: x = {x}")
    return pasos